### PR TITLE
Add Support information in the event source catalog side panel

### PR DESCRIPTION
**Pull request by vikram-raj**
**Story:** https://issues.redhat.com/browse/ODC-6439

**Descriptions:** 
Add information about support on catalog side panel under "Support" Section

**GIF:**
Kapture 2021-12-14 at 14 51 57

### CODE REVIEW

File: frontend/packages/knative-plugin/src/catalog/utils.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  const supportLabel = item.metadata?.labels?.['camel.apache.org/kamelet.support.level'];
+  switch (supportLabel) {
+    case 'Preview': {
+      return 'Tech Preview';
+    }
+    case 'Supported': {
+      return 'Supported';
+    }
+    default: {
+      return 'Community';
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/catalog/utils.ts:13]: I think we would need to translate these.

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: frontend/packages/knative-plugin/src/catalog/utils.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { TFunction } from 'i18next';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+export const getEventSourceSupport = (item: K8sResourceKind, t: TFunction) => {
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/catalog/utils.ts:4]: We could avoid passing `t: TFunction` as a second argument and straight away import and use i18next.t
i.e i18next.t('knative-plugin~Tech Preview')

File: frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const supportText = getEventSourceSupport(kameletsData[3], t);
+    expect(supportText).toBe('Community');
+  });
+
+  it('should return Supported support', () => {
+    const supportText = getEventSourceSupport(kameletsData[1], t);
+    expect(supportText).toBe('Supported');
+  });
+
+  it('should return Supported support', () => {
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts:16]:
```suggestion
  it('should return Tech Preview support', () => {
```

File: frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    const supportText = getEventSourceSupport(kameletsData[0], t);
+    expect(supportText).toBe('Tech Preview');
+  });
+
+  it('should return Supported support', () => {
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts:21]:
```suggestion
  it('should return Community support', () => {
```

File: frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { t } from '../../../../../__mocks__/i18next';
+import { getEventSourceSupport } from '../utils';
+import { kameletsData } from './eventSourceKamelets-data';
+
+describe('getEventSourceSupport', () => {
+  it('should return Community support', () => {
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts:6]: nit
```suggestion
  it('should return Community support if the label is not present', () => {
```

**Review by karthikjeeyar — COMMENTED** [review]
Overall looks good, I have added few nits. It would be helpful if you can add a sample Kamlet CR in the PR description.

File: frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { getEventSourceSupport } from '../utils';
+import { kameletsData } from './eventSourceKamelets-data';
+
+describe('getEventSourceSupport', () => {
+  it('should return Community support if the label is not present', () => {
+    const supportText = getEventSourceSupport(kameletsData[3]);
+    expect(supportText).toBe('Community');
+  });
+
+  it('should return Supported support', () => {
```
Comment (debsmita1) [inline frontend/packages/knative-plugin/src/catalog/__tests__/utils.spec.ts:10]:
```suggestion
  it('should return `Supported` support type', () => {
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by debsmita1** [comment]
verified the changes locally, looks good to me

**Comment by vikram-raj** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
@vikram-raj Is this field for support information needed for all catalog items? If not then there's a way to send extra details panel information using `details` property of `CatalogItem`. Check helm catalog items how it is sending extra information to details panel properties. That should solve this issue without introducing a new property to CatalogItem.

**Comment by vikram-raj** [comment]
@rohitkrai03 I thought the same to make use of `details` property. But the requirement is a bit different here, currently, we have `Support` property but we can only add a link using `supportURL` property. But in this story, we need plain text under `Support`. And if I make use of the `details` property then we have a duplicate entry for `Support` title.

**Comment by rohitkrai03** [comment]
> @rohitkrai03 I thought the same to make use of `details` property. But the requirement is a bit different here, currently, we have `Support` property but we can only add a link using `supportURL` property. But in this story, we need plain text under `Support`. And if I make use of the `details` property then we have a duplicate entry for `Support` title.

@vikram-raj You can send any custom detail property for event sources only. We do that for helm charts. It could be simple string, url or even a react node. SupportURL is something that is supported by the catalog by default but the custom properties can be used for anything else.

**Comment by vikram-raj** [comment]
@rohitkrai03 Updated the PR, PTAL again.

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Comment by jerolimov** [comment]
> /label qe-approved

Damn. I have done this at the same moment. qe-approved +1 :)

**Pull request merged**